### PR TITLE
Auto generate raw data key

### PR DESCRIPTION
**Pull request by Wenxing-hou**
Fix: #2106

The fix has passed the emu test.

### CODE REVIEW

File: os_stub/spdm_device_secret_lib_sample/raw_data_key.h [hunk 1]
```diff
@@ -0,0 +1,438 @@
+/**
```
Comment (jyao1) [inline os_stub/spdm_device_secret_lib_sample/raw_data_key.h:1]: This file includes data structure definition. It should be c file not h file.

Comment (Wenxing-hou) [reply os_stub/spdm_device_secret_lib_sample/raw_data_key.h:1]: Thanks. I have made the script generate `.c` file automatically.

File: os_stub/spdm_device_secret_lib_sample/raw_data_key.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+extern uint8_t m_libspdm_ec384_requester_private_key[48];
+extern uint8_t m_libspdm_ec384_requester_public_key[96];
+
+extern uint8_t m_libspdm_ec521_requester_private_key[66];
+extern uint8_t m_libspdm_ec521_requester_public_key[132];
+#endif /*LIBSPDM_ECDSA_SUPPORT*/
+
+#if (LIBSPDM_RSA_SSA_SUPPORT) || (LIBSPDM_RSA_PSS_SUPPORT)
+extern uint8_t m_libspdm_rsa2048_res_n[257];
```
Comment (jyao1) [inline os_stub/spdm_device_secret_lib_sample/raw_data_key.h:28]: I dont think you can hardcode 257 ort 256.

File: os_stub/spdm_device_secret_lib_sample/raw_data_key.c [hunk 3]
```diff
@@ -0,0 +1,437 @@
+/**
+ *  Copyright Notice:
```
Comment (jyao1) [inline os_stub/spdm_device_secret_lib_sample/raw_data_key.c:2]: please change this to h file.

Comment (Wenxing-hou) [reply os_stub/spdm_device_secret_lib_sample/raw_data_key.c]: Thanks. I have changed the file type.

File: os_stub/spdm_device_secret_lib_sample/raw_data_key.h [hunk 4]
```diff
@@ -0,0 +1,442 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2021-2022 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
+ **/
+
+/*automatic generated by raw_data_key_gen.py*/
+
+#include "spdm_device_secret_lib_internal.h"
```
Comment (jyao1) [inline os_stub/spdm_device_secret_lib_sample/raw_data_key.h:9]: This should be after ifndef.

File: os_stub/spdm_device_secret_lib_sample/raw_data_key.h [hunk 5]
```diff
@@ -0,0 +1,442 @@
...
+ *  Copyright Notice:
+ *  Copyright 2021-2022 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
+ **/
+
+/*automatic generated by raw_data_key_gen.py*/
+
+#include "spdm_device_secret_lib_internal.h"
+
+#ifndef LIBSPDM_RAW_DATA_KEY
```
Comment (jyao1) [inline os_stub/spdm_device_secret_lib_sample/raw_data_key.h:11]: it should be RAW_DATA_KEY_H, to match the file name.

Comment (Wenxing-hou) [reply os_stub/spdm_device_secret_lib_sample/raw_data_key.h]: Thanks. I got it. I will remember the rule.

**Review by jyao1 — APPROVED** [review]

**Pull request merged**